src/RustyObjectStore.jl

```diff
@@ -81,52 +98,193 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
+
+        params = copy(opts.params)
+
+        params["azure_storage_account_name"] = storage_account_name
+        params["azure_container_name"] = container_name
+
+        if !isnothing(storage_account_key)
+            params["azure_storage_account_key"] = storage_account_key
+        end
+
+        if !isnothing(storage_sas_token)
+            params["azure_storage_sas_token"] = storage_sas_token
+        end
+
+        if !isnothing(host)
+            params["azurite_host"] = host
+        end
+
+        return new(Config("az://$(container_name)/", params))
     end
 end
-function Base.show(io::IO, credentials::AzureCredentials)
-    print(io, "AzureCredentials("),
-    print(io, repr(credentials.account), )
-    print(io, ", ", repr(credentials.container))
-    print(io, ", ", "\"*****\"") # don't print the secret key
-    !isempty(credentials.host) && print(io, ", ", repr(credentials.host))
+
+as_config(wrapper::AzureConfig) = wrapper.config
+
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `region::String`: AWS S3 region.
+- `bucket_name::String`: AWS S3 bucket name.
+- `access_key_id::Option{String}`: (Optional) AWS S3 access key id.
+- `secret_access_key::Option{String}`: (Optional) AWS S3 secret access key.
+- `session_token::Option{String}`: (Optional) AWS S3 session_token.
+- `host::Option{String}`: (Optional) Alternative S3 host. For example, if using Minio.
+- `opts::ClientOptions`: (Optional) Client configuration options.
+"""
+struct AwsConfig <: AsConfig
+    config::Config
+    function AwsConfig(;
+        region::String,
+        bucket_name::String,
+        access_key_id::Option{String} = nothing,
+        secret_access_key::Option{String} = nothing,
+        session_token::Option{String} = nothing,
+        host::Option{String} = nothing,
+        opts::ClientOptions = ClientOptions()
+    )
+        params = copy(opts.params)
+
+        params["region"] = region
+        params["bucket_name"] = bucket_name
+
+        if !isnothing(access_key_id)
+            params["aws_access_key_id"] = access_key_id
+        end
+
+        if !isnothing(secret_access_key)
+            params["aws_secret_access_key"] = secret_access_key
+        end
+
+        if !isnothing(session_token)
+            params["aws_session_token"] = session_token
+        end
+
+        if !isnothing(host)
+            params["minio_host"] = host
+        end
+
+        return new(Config("s3://$(bucket_name)/", params))
+    end
+end
+
+as_config(wrapper::AwsConfig) = wrapper.config
+
+function Base.show(io::IO, config::Config)
+    dict = copy(JSON3.read(config.config_string))
+    print(io, "Config(")
+    first = true
+    for key in keys(dict)
+        if first
+            first = false
+        else
+            print(io, ", ")
+        end
+        if occursin(r"secret|token|key", string(key))
+            print(io, repr(key), " => ", repr("*****"))
+        else
+            print(io, repr(key), " => ", repr(dict[key]))
+        end
+    end
```

Comment:
This `show` message doesn't round-trip, which is unfortunate.

it's very helpful to have `show` messages that display objects as valid Julia code for reconstructing an object.

in general, the string representation should parse to valid Julia code which when evaluated produces something equal to the original object, i.e. we should have `x == eval(Meta.parse(repr(x)))`

(potentially with additional "pretty-printing" method, defined by a 3-argument `show` method, but i don't think we need those here)

In practice this is useful for debugging... if you have some object `az`, and want to create a similar object (but don't know how), the natural thing to do is copy-paste its string representation
```julia
julia> az
AzureConfig(Config(:max_retries => "10", :url => "az://my_container/", :retry_timeout_secs => "150", :timeout => "30s", :azure_container_name => "my_container", :connect_timeout => "5s", :azure_storage_account_name => "my_account"))

julia> AzureConfig(Config(:max_retries => "10", :url => "az://my_container/", :retry_timeout_secs => "150", :timeout => "30s", :azure_container_name => "my_container", :connect_timeout => "5s", :azure_storage_account_name => "my_account"))
ERROR: MethodError: no method matching Config(::Pair{Symbol, String}, ::Pair{Symbol, String}, ::Pair{Symbol, String}, ::Pair{Symbol, String}, ::Pair{Symbol, String}, ::Pair{Symbol, String}, ::Pair{Symbol, String})
```

Reply:
How does that work with the masked secrets? Would we only have that for the pretty-print method? If we log something does it use show or pretty-print?

Reply:
unless you define a specific `show(::IO, ::MIME"text/plain", x::MyObject)` method, then it falls back to `show(::IO, x::MyObject)` 

with secrets, we can just return the string "***" or whatever, which means we won't have equality, but we should nevertheless be able to construct a similar object from the show method i.e. show should be a valid constructor (ideally).

i do tend to find this property helpful in debugging, which is why i mention it (rather than it just being aesthetic preferences)